### PR TITLE
INTMDB-521: AWS Secrets Manager to Auth into Terraform Atlas Provider

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

INTMDB-521: AWS Secrets Manager to Auth into Terraform Atlas Provider

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Comment by Zuhairahmed** [comment]
@martinstibbe can you resolve fails in GitHub build and linting tests?

File: website/docs/index.html.markdown [hunk 1]
```diff
@@ -75,6 +76,33 @@ As an alternative to `MONGODB_ATLAS_PUBLIC_KEY` and `MONGODB_ATLAS_PRIVATE_KEY`
...
+
+In order to enable the Terraform MongoDB Atlas Provider to use AWS SM, first create Atlas API Keys and add them as a secret to AWS SM with a basic key with a raw value. See below example:  
+``` 
+     {
+      "public_key": "iepubky",
+      "private_key":"prvkey"
+     }
+```
+
+Next, add assume_role block wih `role_arn`, `secret_name`, and AWS `region` to match the AWS region where secret is stored with AWS SM. See below example:
```
Comment (Zuhairahmed) [inline website/docs/index.html.markdown]:
```suggestion
Next, add assume_role block with `role_arn`, `secret_name`, and AWS `region` to match the AWS region where secret is stored with AWS SM. See below example:
```

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -75,6 +76,33 @@ As an alternative to `MONGODB_ATLAS_PUBLIC_KEY` and `MONGODB_ATLAS_PRIVATE_KEY`
...
+# Configure the MongoDB Atlas Provider to Authenticate with AWS Secrets Manager 
+provider "mongodbatlas" {
+  assume_role {
+    role_arn = "arn:aws:iam::476xxx451:role/mdbsts"
+  }
+  secret_name           = "mongodbsecret"
+  aws_access_key_id     = "ASIXXBNEK"
+  aws_secret_access_key = "ZUZgVb8XYZWEXXEDURGFHFc5Au"
+  aws_session_token     = "IQoXX3+Q="
+  region                = "us-east-2"
```
Comment (Zuhairahmed) [inline website/docs/index.html.markdown:101]:
```suggestion
  region                = "us-east-2"
  sts_endpoint          = "https://sts.us-east-2.amazonaws.com/"
```

**Review by Zuhairahmed — APPROVED** [review]
LGTM! Thanks for making additional edits per Slack @martinstibbe!

File: mongodbatlas/provider.go [hunk 3]
```diff
@@ -226,9 +287,81 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
...
+			case secretsmanager.ErrCodeDecryptionFailure:
+				fmt.Println(secretsmanager.ErrCodeDecryptionFailure, aerr.Error())
+			case secretsmanager.ErrCodeInternalServiceError:
+				fmt.Println(secretsmanager.ErrCodeInternalServiceError, aerr.Error())
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
```
Comment (swattyT) [inline mongodbatlas/provider.go]: Could you remove this comment?

File: mongodbatlas/provider.go [hunk 4]
```diff
@@ -391,3 +524,169 @@ func HashCodeString(s string) int {
...
+					Description:  "IAM Policy JSON describing further restricting permissions for the IAM Role being assumed.",
+					ValidateFunc: validation.StringIsJSON,
+				},
+				"policy_arns": {
+					Type:        schema.TypeSet,
+					Optional:    true,
+					Description: "Amazon Resource Names (ARNs) of IAM Policies describing further restricting permissions for the IAM Role being assumed.",
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+						//ValidateFunc: verify.ValidARN,
```
Comment (swattyT) [inline mongodbatlas/provider.go]: Is this required?

File: mongodbatlas/provider.go [hunk 5]
```diff
@@ -391,3 +524,169 @@ func HashCodeString(s string) int {
...
+					},
+				},
+				"role_arn": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Amazon Resource Name (ARN) of an IAM Role to assume prior to making API calls.",
+					//ValidateFunc: verify.ValidARN,
```
Comment (swattyT) [inline mongodbatlas/provider.go]: Similar to the above, is this required or can be removed?

**Review by swattyT — APPROVED** [review]
LGTM!

File: mongodbatlas/provider.go (hunk 5 above)
Comment (martinstibbe) [reply mongodbatlas/provider.go]: Removed

File: mongodbatlas/provider.go (hunk 4 above)
Comment (martinstibbe) [reply mongodbatlas/provider.go]: Removed

File: mongodbatlas/provider.go (hunk 3 above)
Comment (martinstibbe) [reply mongodbatlas/provider.go]: Removed

**Pull request merged**